src/commands/org/list/metadata.ts

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright (c) 2020, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+import * as fs from 'fs';
+import { Messages } from '@salesforce/core';
+import { FileProperties, ListMetadataQuery } from 'jsforce/api/metadata';
+import {
+  Flags,
+  loglevel,
+  orgApiVersionFlagWithDeprecations,
+  requiredOrgFlagWithDeprecations,
+  SfCommand,
+} from '@salesforce/sf-plugins-core';
+
+Messages.importMessagesDirectory(__dirname);
+const messages = Messages.loadMessages('@salesforce/plugin-org', 'list.metadata');
+
+export type ListMetadataCommandResult = FileProperties[];
+
+export class ListMetadata extends SfCommand<ListMetadataCommandResult> {
+  public static readonly summary = messages.getMessage('description');
+  public static readonly description = messages.getMessage('description');
+  public static readonly examples = messages.getMessages('examples');
+  public static readonly aliases = ['force:mdapi:listmetadata'];
+  public static readonly deprecateAliases = true;
+  public static readonly flags = {
+    'api-version': orgApiVersionFlagWithDeprecations,
+    loglevel,
+    'target-org': requiredOrgFlagWithDeprecations,
+    'output-file': Flags.file({
+      aliases: ['resultfile'],
+      deprecateAliases: true,
+      char: 'f',
+      summary: messages.getMessage('flags.resultfile'),
+    }),
+    'metadata-type': Flags.string({
+      aliases: ['metadatatype'],
+      deprecateAliases: true,
+      char: 'm',
+      summary: messages.getMessage('flags.metadatatype'),
+      description: messages.getMessage('flagsLong.metadatatype'),
+      required: true,
+    }),
+    folder: Flags.string({
+      summary: messages.getMessage('flags.folder'),
+      description: messages.getMessage('flagsLong.folder'),
+    }),
+  };
+
+  public async run(): Promise<ListMetadataCommandResult> {
+    const { flags } = await this.parse(ListMetadata);
+    const conn = flags['target-org'].getConnection(flags['api-version']);
```

Comment:
in plugin-source this flag was only passed to the `conn.metadata.list` method:
https://github.com/salesforcecli/plugin-source/blob/main/src/commands/force/mdapi/listmetadata.ts#L71

and I think the conn version is being picked from your config values/whatever the default was in sfdxCommand.

Prob don't want to pass it to `org.getConnection` as it may cause a mismatch in apiVersions if users had local/global values.

Reply:
With `SfCommand`'s calling `.getConnection` is now recommended to pass the api-version flag, even if undefined. I believe that regardless of what version the Connection is created with, it will be able to call `list` with the correct version